### PR TITLE
Fix parsing of `set_verification_key` permission from mainnet

### PR DESCRIPTION
**Pull request by mrmr1993**
This PR allows for the parsing of the `set_verification_key` permission format that's emitted by mainnet/`compatible` nodes. This allows us to import the accounts dumped by the `fork_config` GraphQL query.

### CODE REVIEW

**Comment by mrmr1993** [comment]
!ci-build-me

File: src/lib/runtime_config/runtime_config.ml [hunk 1]
```diff
@@ -157,6 +157,13 @@ module Json_layout = struct
         module Verification_key_perm = struct
           type t = { auth : Auth_required.t; txn_version : Txn_version.t }
           [@@deriving dhall_type, sexp, yojson, bin_io_unversioned]
+
+          let of_yojson = function
+            | `String _ as json ->
+                let%map.Result auth = Auth_required.of_yojson json in
+                { auth; txn_version = Txn_version.current }
```
Comment (georgeee) [inline src/lib/runtime_config/runtime_config.ml:164]:
```suggestion
                { auth; txn_version = 1 }
```

Technically, setting to current version is not a correct way

**Review by georgeee — COMMENTED** [review]
Do we really need this change?

We have a script that extracts runtime config + ledger `.tar.gz` DBs for Berkeley out of fork config extracted from mainnet. Couldn't we just add one line to that script and keep the logic simpler here in the Ocaml code?

The code line for the script would be something as simple as:
```sh
jq '.ledger.accounts = [.ledger.accounts[] | if .permissions.set_verification_key == "signature" then .permissions.set_verification_key = {auth:"signature", txn_version: "1"} else . end ]'
```

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
Force-pushed to remove an accidental merge.

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
This was overruled in favour of a more-complex bash-driven migration script (see https://github.com/MinaProtocol/mina/pull/15003). Closing.

**Pull request closed without merging**